### PR TITLE
feat(action-group, block, panel): add `menuPlacement` and `menuFlipPlacements` properties

### PR DESCRIPTION
**Pull request by driskull**
**Related Issue:** #7516

## Summary

- add `menuFlipPlacements` property
- add `menuPlacement` property
- add tests
- update stories

### CODE REVIEW

File: packages/calcite-components/src/components/block/block.tsx [hunk 1]
```diff
@@ -94,6 +94,11 @@ export class Block
    */
   @Prop({ reflect: true }) dragHandle = false;
 
+  /**
+   * Defines the available placements that can be used when a flip occurs.
```
Comment (macandcheese) [inline packages/calcite-components/src/components/block/block.tsx:98]: Do we want a bit more context here for "a flip occurs" - maybe something about the rendered action menu flipping? It makes sense to me as someone who knows it refers to internally rendered action menu, but maybe these could be confusing without a more descriptive description (or even prop name)? Applies to others.

Comment (driskull) [reply packages/calcite-components/src/components/block/block.tsx:98]: Yeah probably. I copied this from another component.

cc @geospatialem @DitwanP on doc

Comment (macandcheese) [reply packages/calcite-components/src/components/block/block.tsx:98]: I guess my comment was more about _what_ is being affected by this. In this case, it's "the action menu that is rendered when you slot actions in ___"

"... the component's fallback `placement`" still isn't clear to a user without context - maybe "Specifies the fallback `placement` of the component's (internally rendered Action Menu that displays when slotted actions overflow)" (but ... less bad verbiage)

Comment (driskull) [reply packages/calcite-components/src/components/block/block.tsx:98]: @geospatialem combobox is out of scope I think. It already has flipPlacements just not placement.

Popover, input-date-picker, split-button, and dropdown already has these properties.

Comment (geospatialem) [reply packages/calcite-components/src/components/block/block.tsx]: Added a new PR https://github.com/Esri/calcite-design-system/pull/10261 to cover the remaining components for consistency.

File: packages/calcite-components/src/components/block/resources.ts [hunk 2]
```diff
@@ -43,3 +43,5 @@ export const ICONS = {
   valid: "check-circle",
   invalid: "exclamation-mark-triangle",
 } as const;
+
+export const defaultMenuPlacement = "bottom-end";
```
Comment (jcfranco) [inline packages/calcite-components/src/components/block/resources.ts:47]: There’s a [`defaultMenuPlacement` in `floating-ui.ts`](https://github.com/Esri/calcite-design-system/blob/dev/packages/calcite-components/src/utils/floating-ui.ts#L273) as well. How should we decide when to use it vs creating a separate one?

Comment (driskull) [reply packages/calcite-components/src/components/block/resources.ts:47]: They are both positioned bottom but for dropdown/combobox bottom-start makes more sense whereas for action-menu's mostly positioned at the end bottom-end makes more sense. 

I can put this const in floating-ui.ts

**Comment by driskull** [comment]
@jcfranco can you review again?

File: packages/calcite-components/src/components/action-group/action-group.tsx [hunk 3]
```diff
@@ -95,6 +95,16 @@ export class ActionGroup
...
 
+  /**
+   * Specifies the component's fallback menu `placement` when it's initial or specified `placement` has insufficient space available.
+   */
+  @Prop() menuFlipPlacements: FlipPlacement[];
+
+  /**
+   * Determines where the action menu will be positioned.
+   */
+  @Prop({ reflect: true }) menuPlacement: LogicalPlacement;
```
Comment (driskull) [inline packages/calcite-components/src/components/action-group/action-group.tsx:106]: Do we prefer `actionMenuPlacement` or is `menuPlacement` sufficient?

Comment (jcfranco) [reply packages/calcite-components/src/components/action-group/action-group.tsx:106]: I think the current name is clear and concise.

File: packages/calcite-components/src/tests/commonTests/floatingUI.ts [hunk 4]
```diff
@@ -132,3 +132,38 @@ export async function delegatesToFloatingUiOwningComponent(
     expect(await floatingUiOwningComponent.getProperty("overlayPositioning")).toBe("fixed");
   });
 }
+
+/**
+ * Helper to test if a component has a calcite-action-menu wired up correctly with placement and flipPlacements.
```
Comment (jcfranco) [inline packages/calcite-components/src/tests/commonTests/floatingUI.ts:137]: ✨🏆✨ This is great, thanks for adding it!

File: packages/calcite-components/src/components/block/block.tsx (hunk 1 above)
Comment (jcfranco) [reply packages/calcite-components/src/components/block/block.tsx]: For consistency and readability, would we want to consider updating similar properties to use the `menu` prefix? I don't recall if we have properties other than `overlayPositioning`? 🤔

**Review by jcfranco — APPROVED** [review]
💭💭💭💭💭💭💭💭💭💭💭💭💭💭💭💭💭💭💭💭💭💭💭💭💭💭💭💭💭💭💭💭
💭💭↔️↔️↔️💭↔️💭💭💭💭💭↔️💭↔️↔️↔️↔️💭↔️↔️↔️↔️💭↔️↔️↔️↔️↔️💭↔️💭
💭↔️💭💭💭💭↔️💭💭💭💭💭↔️💭↔️💭💭💭💭↔️💭💭💭💭💭💭↔️💭💭💭↔️💭
💭💭↔️↔️💭💭↔️💭💭↔️💭💭↔️💭↔️↔️↔️💭💭↔️↔️↔️💭💭💭💭↔️💭💭💭↔️💭
💭💭💭💭↔️💭↔️💭↔️💭↔️💭↔️💭↔️💭💭💭💭↔️💭💭💭💭💭💭↔️💭💭💭💭💭
💭↔️↔️↔️💭💭💭↔️💭💭💭↔️💭💭↔️↔️↔️↔️💭↔️↔️↔️↔️💭💭💭↔️💭💭💭↔️💭
💭💭💭💭💭💭💭💭💭💭💭💭💭💭💭💭💭💭💭💭💭💭💭💭💭💭💭💭💭💭💭💭

**Comment by driskull** [comment]
> For consistency and readability, would we want to consider updating similar properties to use the menu prefix? I don't recall if we have properties other than overlayPositioning

Yes, potentially. This would be a breaking change though. 🔧 

Should we have a followup issue? cc @DitwanP @geospatialem

**Comment by DitwanP** [comment]
> Should we have a followup issue? cc @DitwanP @geospatialem

[Follow-up Issue](https://github.com/Esri/calcite-design-system/issues/10391)

**Pull request merged**